scripts/runtestsuite/runtestsuite.py

```diff
@@ -117,6 +117,41 @@ def write_rgba_png_file(filepath, rows):
     with open(filepath, 'wb') as file:
         writer.write(file, rows)
 
+def git_installed(directory):
+    temp = 'git -C {0} --version'.format(directory)
+    print(temp)
+    return command_is_valid('git -C {0} --version'.format(directory))
+
+def under_git_control(directory):
+    return command_is_valid('git -C {0} rev-parse --is-inside-work-tree'.format(directory))
+
+def get_git_hash(directory):
+    return command_output('git -C {0} rev-parse HEAD'.format(directory))
+
+def get_git_title(directory):
+    return command_output('git -C {0} rev-parse --abbrev-ref HEAD'.format(directory))
+
+def command_output(command):
+    output = "N/A"
+    try:
+        output = subprocess.check_output(command.split())
+    except OSError:
+        pass
+    return output
+
+def command_is_valid(command):
+    is_valid = 1
+    try:
+        is_valid = subprocess.check_call(
+            command.split(),
+            stdin = subprocess.PIPE, 
+            stdout = open(os.devnull, 'wb'),
+            stderr = subprocess.STDOUT)
+    except (subprocess.CalledProcessError, OSError):
+        pass
+    
+    return is_valid == 0
```

Comment:
As far as I can tell, `subprocess` doesn't read spaces properly. Anything separated by a space, needs to be separate elements of an array. 

I'm specifying the standard output and error so they don't get displayed in the terminal (and disrupt the nice table that gets printed). It is possible that this would be cleaner if I used another `subprocess` method here, I'll have to investigate.

Reply:
Ok, good to know ! We can keep this, but maybe you can mention these important details directly in the code ?